core/modules/map.rs

```diff
@@ -55,46 +61,79 @@ pub(crate) enum SymbolicModule {
   Mod(ModuleId),
 }
 
+struct DynImportModEvaluate {
+  load_id: ModuleLoadId,
+  module_id: ModuleId,
+  promise: v8::Global<v8::Promise>,
+  module: v8::Global<v8::Module>,
+}
+
 /// A collection of JS modules.
 pub(crate) struct ModuleMap {
-  // Handling of specifiers and v8 objects
-  pub handles: Vec<v8::Global<v8::Module>>,
-  pub info: Vec<ModuleInfo>,
-  pub(crate) by_name_js: HashMap<ModuleName, SymbolicModule>,
-  pub(crate) by_name_json: HashMap<ModuleName, SymbolicModule>,
-  pub(crate) next_load_id: ModuleLoadId,
-
   // Handling of futures for loading module sources
-  pub loader: Rc<dyn ModuleLoader>,
-  pub(crate) dynamic_import_map:
-    HashMap<ModuleLoadId, v8::Global<v8::PromiseResolver>>,
-  pub(crate) preparing_dynamic_imports:
-    FuturesUnordered<Pin<Box<PrepareLoadFuture>>>,
-  pub(crate) pending_dynamic_imports:
-    FuturesUnordered<StreamFuture<RecursiveModuleLoad>>,
-
-  // This store is used temporarily, to forward parsed JSON
-  // value from `new_json_module` to `json_module_evaluation_steps`
+  // TODO(mmastrac): we should not be swapping this loader out
+  pub(crate) loader: RefCell<Rc<dyn ModuleLoader>>,
```

Comment:
I don't understand this change... Why is it `RefCell<Rc<_>>`? This seems very fishy

Reply:
We need to swap out loader during extension loads because of how things are currently done. It's definitely fishy and I'd like to fix that at some point.

Originally we just used the outer RefCell to swap it out, but that's gone.

Reply:
I see. That makes sense. Could you update a comment and remove `pub(crate)` and then add `get_loader()` (public) and `set_loader()` that only `deno_core` can use?